proposals/0000-Move-Platforms-to-Own-Packages.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Move iOS and Android Specific Code to Their Own Packages
+author:
+- Ian Meikle
+date: today
+---
+
+# RFC0000: Move iOS and Android Specific Code to Their Own Packages
+
+## Summary
+
+Re-organize the React Native repo as a Yarn workspace monorepo, and move platform specific JS and native code for iOS and Android into their own packages.
+
+## Basic example
+
+Common APIs and components would remain in `react-native`:
+
+```js
+import {AppRegistry, Alert, View} from 'react-native';
+```
+
+Platform-specific components and APIs would be imported from the platform's module:
```

Comment:
What about things that should be cross platform but just haven’t been implemented on the other platform yet? Do they start in a platform specific package and move into the core package later?

Or is it more based on the particular api/component in which case there would still be some things in core that don’t work on both platforms?

Reply:
> What about things that should be cross platform but just haven’t been implemented on the other platform yet?

I was thinking that an `UnimplementedView` stub should be in the main package and an implemented JS file would be in the platform module.

Like in the case of `CheckBox`, a `UnimplementedView` stub `CheckBox.js` would be in the main package, and a `CheckBox.android.js` in `react-native-android`

Reply:
I think the core should be slim enough that its pretty much required to have it all implemented.  So for instance, checkbox wouldn't be part of core.

Reply:
Agreed, would be good to keep core to a minimum "This is what you need to implement to be considered a full React Native implementation". Other stuff could be implemented by the platform implementers by submitting pull requests to the Slimmen'd repos (like what will probably happen with Windows' WebView implementation)